airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py

```diff
@@ -208,21 +217,34 @@ def _filter_dynamic_fields(self, records: Iterable) -> Iterable:
                         record["properties"].pop(key)
             yield record
 
+    def _transform(self, records: Iterable) -> Iterable:
+        yield from records
+
+    @staticmethod
+    def _field_to_datetime(value: Union[int, str]) -> pendulum.datetime:
+        if isinstance(value, int):
+            value = pendulum.from_timestamp(value / 1000.0)
+        elif isinstance(value, str):
+            value = pendulum.parse(value)
+        else:
+            raise ValueError(f"Unsupported type of datetime field {type(value)}")
+        return value
+
     def _filter_old_records(self, records: Iterable) -> Iterable:
         """Skip records that was updated before our start_date"""
         for record in records:
-            if self.updated_field:
-                updated_at = record[self.updated_field]
-                if isinstance(updated_at, int):
-                    updated_at = pendulum.from_timestamp(updated_at / 1000.0)
-                elif isinstance(updated_at, str):
-                    updated_at = pendulum.parse(updated_at)
-                else:
-                    raise ValueError(f"Unsupported type of update cursor {type(updated_at)}")
+            updated_at = self._record_bookmark(record)
+            if updated_at:
+                updated_at = self._field_to_datetime(updated_at)
                 if updated_at < self._start_date:
                     continue
             yield record
 
+    def _record_bookmark(self, record: Mapping[str, Any]) -> Any:
```

Comment:
why is this method in the base stream class and not in the incremental stream class?

Reply:
because `start_date` functionality implemented through it, we filter records before emitting